src/lib.rs

```diff
@@ -1 +1,139 @@
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    directory,
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            } else {
+                windows::Win32::System::Threading::CreateProcessW(
+                    PWSTR::default(),
+                    command,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    std::ptr::null() as *const SECURITY_ATTRIBUTES,
+                    inherit_handles,
+                    process_creation_flags,
+                    std::ptr::null() as *const c_void,
+                    PWSTR::default(),
+                    &si,
+                    &mut pi as *mut PROCESS_INFORMATION,
+                )
+            };
+
+            if res.as_bool() {
+                Ok(Self {
+                    command: command.to_string(),
+                    process_information: pi,
+                })
+            } else {
+                Err(ChildProcessError::CreationFailed(format!(
+                    "{:?}",
+                    GetLastError()
+                )))
+            }
+        }
+    }
+
+    pub fn wait(&self) -> ExitStatus {
+        unsafe {
+            let exit_code = WaitForSingleObject(self.process_information.hProcess, INFINITE);
+            CloseHandle(self.process_information.hProcess);
+            CloseHandle(self.process_information.hThread);
+
+            ExitStatus(exit_code)
+        }
+    }
+
+    pub fn try_wait(&self, duration: u32) -> Result<ExitStatus, ExitStatusError> {
+        unsafe {
+            match WaitForSingleObject(self.process_information.hProcess, duration) {
+                0 => Ok(ExitStatus(0)),
+                _ => Err(ExitStatusError::WaitFailed(format!("{:?}", GetLastError()))),
+            }
+        }
+    }
+
+    pub fn kill(&self) -> Result<(), ChildProcessError> {
+        unsafe {
+            if TerminateProcess(self.process_information.hProcess, 0).as_bool() {
+                Ok(())
+            } else {
+                Err(ChildProcessError::KillFailed(format!(
+                    "{:?}",
+                    GetLastError()
+                )))
+            }
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum ChildProcessError {
+    #[error("cannot create childprocess: {0}")]
+    CreationFailed(String),
+    #[error("cannot kill process: {0}")]
+    KillFailed(String),
+}
+
+pub struct ExitStatus(u32);
+
+impl ExitStatus {
+    pub fn success(&self) -> bool {
+        self.0 == 0
+    }
+
+    pub fn code(&self) -> u32 {
+        self.0
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum ExitStatusError {
+    #[error("time-out elapsed ({0})")]
+    WaitTimeout(u32),
+    #[error("wait abandoned ({0})")]
+    WaitAbandoned(u32),
+    #[error("failed to wait: {0}")]
+    WaitFailed(String),
+}
```

Comment:
Oh you have 2 error types... I don't think you need 2 error types in this crate.

Use only one and name it `Error`. Then make a public type alias for `Result`.

```rust
enum Error { /* ... */ }

type Result<T> = std::result::Result<T, Error>;
```

Then you won't need to give the error type anymore inside your library.

This is a common pattern for Rust library and it is inspired from std. Here is std::io::Result: https://doc.rust-lang.org/std/io/type.Result.html#

You can see everywhere here that only one type argument is used in the result because it is using that alias: https://doc.rust-lang.org/std/process/struct.Child.html